Source/GTMSessionFetcher.m

```diff
@@ -3749,21 +3753,30 @@ - (void)setCallbackQueue:(nullable dispatch_queue_t)queue {
  * run after the completion handler. Serial dispatch is guaranteed by dispatching to an intermediate
  * serial queue that targets the actual callback queue (which may or may not be serial).
  */
-- (nonnull dispatch_queue_t)serialCallbackQueue {
+- (nonnull dispatch_queue_t)internalCallbackQueue {
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
-    return [self serialCallbackQueueUnsynchronized];
+    return [self internalCallbackQueueUnsynchronized];
   }  // @synchronized(self)
 }
 
-- (nonnull dispatch_queue_t)serialCallbackQueueUnsynchronized {
+- (nonnull dispatch_queue_t)internalCallbackQueueUnsynchronized {
   GTMSessionCheckSynchronized(self);
 
-  if (!_serialCallbackQueue) {
-    _serialCallbackQueue = SerialCallbackQueueForTargetQueue(_callbackQueue);
+  if (!_internalCallbackQueue) {
+    // If the fetcher was provided blocks for passive sequential callbacks (not ones where the
+    // system waits for a response, such as the response callback), create a serial queue wrapper
+    // targeting the client-provided callback queue.
```

Comment:
Maybe expand this comment that we do this to help out developers by ensuring they won't have to deal with out of order messages if they failed to make their custom queue serial.

Reply:
Done.